.github/workflows/package-manager-ci.yml

```diff
@@ -0,0 +1,92 @@
+name: package-manager-ci
+on:
+  push:
+    paths-ignore:
+        - 'docs/**'
+        - '*.md'
+  pull_request:
+    paths-ignore:
+        - 'docs/**'
+        - '*.md'
+jobs:
+  pnpm:
+    name: pnpm package manager on ${{ matrix.node-version }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-latest, windows-latest]
+        node-version: [12, 14, 16]
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2.4.0
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Use pnpm
+        uses: pnpm/action-setup@v2.0.1
+        with:
+          version: ^6.0.0
+      - name: Install dependancies
+        run: pnpm install
+      - name: Tests
+        run: pnpm run test-ci
+
+  yarn:
+    name: yarn package manager on ${{ matrix.node-version }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-latest]
+        node-version: [12, 14, 16]
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2.4.0
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Use yarn
+        run: |
+          npm install -g yarn
+          yarn set version berry && yarn set version 2.4.2
+          echo "nodeLinker: node-modules" >> .yarnrc.yml
+          # see https://github.com/yarnpkg/berry/issues/2935#issuecomment-911299992
+          yarn add --dev typescript@~4.3.2
+          yarn install
+        env:
+          # needed due the yarn.lock file in pino's .gitignore
+          YARN_ENABLE_IMMUTABLE_INSTALLS: false
+      - name: Tests
+        run: |
+          # yarn doesn't support ESM yet
+          mv ./test/esm/index.test.js ./test/esm/index.test.skip.js
+          yarn run test-ci
+
+  yarn-pnp:
+    name: yarn-pnp package manager on ${{ matrix.node-version }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-latest]
+        node-version: [12, 14, 16]
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2.4.0
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Use yarn
+        run: |
+          npm install -g yarn
+          yarn set version berry
+          echo "nodeLinker: pnp" >> .yarnrc.yml 
+          echo "pnpMode: loose" >> .yarnrc.yml 
```

Comment:
Why are you testing in loose mode? This will allow undeclared dependencies to go through

Reply:
There are some packages that do not have added their packages as dependency or peer ones. One of them is `tap`.
Fixing all of them seems very slow.

I'm trying to track `packageExtensions` but this could be massive actually

There is no option to generates the `packageExtensions` during a `yarn run ..`?

Reply:
The one you currently have for `debug` is unnecessary, it's declared as an optional peer dependency already.

> There are some packages that do not have added their packages as dependency or peer ones. One of them is tap.
Fixing all of them seems very slow.

Sadly there are still packages that don't declare their dependencies correctly but if it's of any consultation it used to be a lot worse

> There is no option to generates the packageExtensions during a yarn run ..?

We haven't made a tool for it yet, no. It's a bit difficult as well since we have no clue what version or what type of dependency it is supposed to be.

Reply:
Removed the loose mode 👍🏽 

The `debug` is needed otherwise I have some warning that let the tests fail